extensions/objects/templates/image-editor.component.ts

```diff
@@ -982,6 +982,17 @@ export class ImageEditorComponent implements OnInit, OnChanges {
       let gifHeight = dimensions.height;
       this.processGIFImage(imageDataURI, gifWidth, gifHeight, null, successCb);
     } else if (mimeType === 'data:image/svg+xml') {
+      const svgString = atob(imageDataURI.split(',')[1]);
+      let domParser = new DOMParser();
+      let doc = domParser.parseFromString(svgString, 'image/svg+xml');
+      doc.querySelectorAll('*').forEach((node) => {
+        if (node.tagName.toLowerCase() === 'svg') {
+          node.removeAttribute('data-name');
+        }
+      });
```

Comment:
Thanks @vojtechjelinek. `DOMParser.parseFromString` return an HTML/XML document and we cannot directly remove any attribute from it since it is a document and an error will be raised `Property 'removeAttribute' does not exist on type 'Document'`. Therefore we loop through all its node and remove `data-name` attribute from them.
Will surely add code comments!

Reply:
But why do we need to remove `data-name`? That is the main thing I want to see in the comment.

Reply:
As mentioned in [assets/constants.ts](https://github.com/oppia/oppia/blob/develop/assets/constants.ts), the codebase allows SVG tag-specific attributes based on the list of tags and attributes specified in this project: [DOMPurify](https://github.com/cure53/DOMPurify). 
Usually, we report the invalid tags and ask the users to fix the SVG and reupload. However, the `data-name` attribute seems to be a lot common in uploaded SVG's and it is a bit inconvenient for the users to remove this one every time. More context can be found [here](https://github.com/oppia/oppia/pull/14876#issuecomment-1031422351).
I have added this explanation in the code. PTAL! @vojtechjelinek